### PR TITLE
Clean up AnyQueryNode code

### PR DESCRIPTION
**Pull request by sabi0**
- removed redundant field initializers
- fixed a typo in the field name: `minimumMatching_m_Elements`
- removed redundant `instanceof` checks
- replaced type casts with a pattern variable
- changed `size() == 0` to `isEmpty()`
- removed unnecessary explicit `toString()` calls

### CODE REVIEW

**Comment by dweiss** [comment]
Thank you, LGTM. Could you add a changes entry as well?

File: lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/AnyQueryNode.java [hunk 1]
```diff
@@ -21,37 +21,29 @@
...
-        if (clause instanceof FieldQueryNode) {
-
-          if (clause instanceof QueryNodeImpl) {
-            ((QueryNodeImpl) clause).toQueryStringIgnoreFields = true;
-          }
-
-          if (clause instanceof FieldableNode) {
-            ((FieldableNode) clause).setField(field);
-          }
+        if (clause instanceof FieldQueryNode qn) {
```
Comment (dweiss) [inline lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/AnyQueryNode.java:37]: This won't backport to 9x, unfortunately. I'll just correct it manually when backporting. Or should we leave this on main only, @uschindler - wdyt?

Comment (sabi0) [reply lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/AnyQueryNode.java:37]: I could undo the pattern variable part, leaving it to #12295.

Comment (uschindler) [reply lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/AnyQueryNode.java:37]: Yes please.

**Review by uschindler — APPROVED** [review]
Looks fine. Thanks for also fixing the typo in field name. 🤩

**Pull request merged**